lib/Common/DataStructures/Comparer.h

```diff
@@ -114,9 +114,11 @@ struct RecyclerPointerComparer
     }
 };
 
+// FNV-1a hash -> https://en.wikipedia.org/wiki/Fowler%E2%80%93Noll%E2%80%93Vo_hash_function
+#define CC_HASH_OFFSET_VALUE 2166136261
 #define CC_HASH_LOGIC(hash, byte) \
-    hash  = _rotl(hash, 7);        \
-    hash ^= byte;
+    hash ^= byte;                 \
+    hash *= 16777619
```

Comment:
Surely multiply is slower. However, when we have a hashCode that doesn't distribute well, we end up having multiple hashCode generations per each lookup because we have lots of collisions. Besides, we loop on WCHAR.. Well, neither of the algorithms planned for that. What would happen to our current algorithm if that WCHAR would carry a unicode? Previous algorithm fails at the beginning due to strict 128 ASCII distance from each possibility block. Popularly, for hash/map, we MOD the hash-key by the number of capacity we have. Old algorithm is being in order (128 ASCII blocks), not surprisingly results in a very bad distribution. i.e. 127 far from 128 more like to end up in 127 from 0-128 while the new algorithm generates has codes more likely to spread across the table. That's why the perf gain. I will do more benchmarks but so far so good. PS2: power of 128 approach collides even faster when the length of the data grows.